sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java

```diff
@@ -655,7 +683,26 @@ private Bound(String name, String filenamePrefix, String filenameSuffix,
        */
       public Bound<T> withFooter(@Nullable String footer) {
         return new Bound<>(name, filenamePrefix, filenameSuffix, header, footer, coder, numShards,
-            shardTemplate, false);
+            shardTemplate, validate, outputWrapperFactory);
+      }
+
+      /**
+       * Returns a transform for writing to text files like this one but that has the given
+       * {@link OutputWrapperFactory} to be used by the {@link FileBasedSink} during output. The
+       * default is value is {@link FileBasedSink.CompressionType#UNCOMPRESSED}.
+       *
+       * <p>
+       * A {@code null} value will reset the value to the default value mentioned above.
+       *
+       * <p>
+       * Does not modify this object.
+       *
+       * @param outputWrapperFactory the {@link OutputWrapperFactory} to be used by the
+       *        {@link FileBasedSink} during output
```

Comment:
please do not indent follow-on lines; as with `<p>` this is to stay stylistically consistent with the rest of the codebase.